pandas/_libs/tslib.pyx

```diff
@@ -1176,12 +1176,18 @@ cdef class _Timestamp(datetime):
 
         If warn=True, issue a warning if nanoseconds is nonzero.
         """
+        cdef:
+            pandas_datetimestruct dts
+            _TSObject ts
+
         if self.nanosecond != 0 and warn:
             warnings.warn("Discarding nonzero nanoseconds in conversion",
                           UserWarning, stacklevel=2)
-        return datetime(self.year, self.month, self.day,
-                        self.hour, self.minute, self.second,
-                        self.microsecond, self.tzinfo)
+        ts = convert_to_tsobject(self, self.tzinfo, None, 0, 0)
```

Comment:
maybe we should add a comment here that `self` is converted to a TSObject for performance reasons (faster attribute access) for future reference, as I agree you wouldn't necessarily think this